src/Compilers/CSharp/Portable/FlowAnalysis/DataFlowPass.cs

```diff
@@ -1721,14 +1721,40 @@ public override BoundNode VisitAssignmentOperator(BoundAssignmentOperator node)
             Assign(node.Left, node.Right, refKind: node.RefKind);
             return null;
         }
-        
+
         public override BoundNode VisitDeconstructionAssignmentOperator(BoundDeconstructionAssignmentOperator node)
         {
             base.VisitDeconstructionAssignmentOperator(node);
 
-            foreach(BoundExpression variable in node.LeftVariables)
+            foreach (BoundExpression variable in node.LeftVariables)
+            {
+                if (variable.Kind == BoundKind.DeconstructionVariables)
+                {
+                    VisitDeconstructionVariables((BoundDeconstructionVariables)variable);
```

Comment:
Doesn't `base.VisitDeconstructionAssignment` do this? Same comment for `VisitDeconstructionVariables` below.